packages/playwright/src/generators/init/utils/update-targets.ts

```diff
@@ -0,0 +1,71 @@
+import {
+    joinPathFragments,
+    ProjectConfiguration,
+    Tree,
+    updateJson,
+} from '@nrwl/devkit';
+
+function updateNxJson(tree) {
+    updateJson(tree, 'nx.json', nxJson => {
+        const updatedNxJson = { ...nxJson };
+
+        if (
+            Array.isArray(
+                nxJson.taskRunnerOptions?.default?.options?.cacheableOperation,
+            )
+        ) {
+            updatedNxJson.taskRunnerOptions.default.options.cacheableOperations.push(
+                'e2e-docker',
+            );
+        }
+
+        return nxJson;
+    });
+}
+
+export function updateProjectJsonWithNativeVisualRegressionTargets(
+    project: ProjectConfiguration,
+    tree: Tree,
+) {
+    updateJson(
+        tree,
+        joinPathFragments(project.root, 'project.json'),
+        projectJson => {
+            const updatedProjectJson = { ...projectJson };
+
+            if (!projectJson.targets['e2e-docker']) {
+                updatedProjectJson.targets['e2e-docker'] = {
+                    executor: 'nx:run-commands',
+                    options: {
+                        commands: [
+                            {
+                                command:
+                                    'docker run -v $PWD:/project -w /project --rm --ipc=host mcr.microsoft.com/playwright:v1.30.0-jammy npx nx run next-custom-server-e2e:e2e',
+                                forwardAllArgs: false,
+                            },
+                        ],
+                    },
+                    configurations: {
+                        updatesnapshots: {
+                            args: '--extra=--update-snapshots',
```

Comment:
Could we also get an additional arg?
--grep @visual-regression
When updating snapshots we wouldn't want to run the entire test suite, rather limit it to only the tests which actual perform visual regression